lib/main.js

```diff
@@ -1,156 +1,98 @@
-      this.worker = new Task(require.resolve('./worker.js'));
-      idleCallbacks.delete(createWorkerCallback);
+  }
+
+  preInitialization(connection) {
+    // connection._onNotification({ method: 'tslint/status' }, (params) => {
+    //   // This notification is used to set the status indicator in VSCode.
+    //
+    //   const status = params.state;
+    //   switch (status) {
+    //     case 1: // ok
+    //       atom.notifications.addSuccess("TSLint is OK");
+    //       break;
+    //     case 2: // warning
+    //       atom.notifications.addWarning("TSLint is not OK");
+    //       break;
+    //     case 3: // error
+    //       atom.notifications.addError("TSLint is not OK");
+    //       break;
+    //     default:
+    //       console.error("TSLint status changed in an unknown way!")
+    //   }
+    // });
+
+    connection._onRequest({ method: 'tslint/noLibrary' }, () => {
+      atom.notifications.addError("TSLint can't find TSLint Library in Project.");
     });
-    idleCallbacks.add(createWorkerCallback);
-  },
-
-  deactivate() {
-    idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID));
-    idleCallbacks.clear();
-    this.subscriptions.dispose();
-
-    workerHelper.terminateWorker();
-    this.worker = null;
-  },
-
-  provideLinter() {
-    return {
-      name: 'TSLint',
-      grammarScopes,
-      scope: 'file',
-      lintOnFly: true,
-      lint: async (textEditor) => {
-        if (this.ignoreTypings && textEditor.getPath().toLowerCase().endsWith('.d.ts')) {
-          return [];
-        }
 
-        if (!this.worker) {
-          await waitOnIdle();
-        }
+    const getConfig = () => ({
+      enable: true,
+      jsEnable: atom.config.get('linter-tslint.javascriptSupport'),
+      rulesDirectory: atom.config.get('linter-tslint.rulesDirectory'),
+      configFile: path.join(this.projectPath, 'tslint.json'),
+      ignoreDefinitionFiles: atom.config.get('linter-tslint.ignoreTypings'),
+      exclude: [''],
+      validateWithDefaultConfig: !atom.config.get('linter.tslint.useLocalTslint'),
+      nodePath: process.execPath,
+      run: 'onType',
+      alwaysShowRuleFailuresAsWarnings: false,
+      autoFixOnSave: atom.config.get('linter-tslint.fixOnSave'),
+      trace: atom.config.get('core.debugLSP') ? { server: 'verbose' } : {},
+    });
 
-        workerHelper.startWorker(this.worker, config);
+    connection._onRequest({ method: 'workspace/configuration' }, getConfig);
 
-        const text = textEditor.getText();
-        const results = await workerHelper.requestJob('lint', textEditor);
+    const configChanged = () => {
+      connection.didChangeConfiguration({ settings: getConfig() });
+    };
 
-        if (textEditor.getText() !== text) {
-          // Text has been modified since the lint was triggered, tell linter not to update
-          return null;
-        }
+    atom.config.observe('linter-tslint.javascriptSupport', configChanged);
```

Comment:
`atom.config.observe` returns a subscription, you should keep track of these all and dispose of them when the package's `deactivate` is called. Look at the original implementation for how to do this.